ui/src/components/ConfigurationForm/ConfigurationForm.component.tsx

```diff
@@ -119,13 +165,53 @@ export const ConfigurationForm: React.FC<ConfigurationFormProps> = ({ isEmbedded
     });
   };
 
+  const [snackbar, setSnackbar] = useState<SnackbarConfig>({
+    open: false,
+    severity: 'info',
+  });
+  const handleSnackClose = (event?: SyntheticEvent | Event, reason?: string) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setSnackbar({ open: false });
+  };
+
+  const handleShareConfiguration = async () => {
+    const casted = form.getValues() as StarterRequest;
+    const urlToShare = stringifyUrl(
+      { url: window.location.href, query: { ...casted } },
+      { skipNull: true, skipEmptyString: true }
+    );
+    await navigator.clipboard.writeText(urlToShare);
+    setSnackbar({ open: true, severity: 'info', message: 'Link to configuration copied to clipboard.' });
+  };
+
   return (
     <Box>
       {!isEmbedded && (
-        <Typography variant="h3" component="h3" fontWeight={300} gutterBottom>
-          Generate tapir project
-        </Typography>
+        <Stack direction="row" spacing={0.25}>
+          <Typography variant="h3" component="h3" fontWeight={300} gutterBottom>
+            Generate tapir project
+          </Typography>
+          <Tooltip title="Share configuration" arrow>
+            <IconButton color="secondary" aria-label="share configuration" onClick={handleShareConfiguration}>
+              <ShareTwoToneIcon />
+            </IconButton>
+          </Tooltip>
+        </Stack>
       )}
+
+      <Snackbar
```

Comment:
There already is one snackbar on this page - in the `ApiCallAddons` component. Maybe it can be reused? Or at least unified, because this one differs a little (f.e. different horizontal position).

Reply:
Right. For now I have aligned positions. I will think about aligning them as here it contains also `x` so that one can clean it up nicely. IMHO it is a subject of the follow-up